pkg/server/remote_storage.go

```diff
@@ -80,7 +80,7 @@ func (s *ImmuServer) initializeRemoteStorage(storage remotestorage.Storage) erro
 			return err
 		}
 
-		if !fileExists(localIdentifierFile) {
+		if !fileExists(localIdentifierFile) || s.Options.RemoteStorageOptions.S3ExternalIdentifier {
```

Comment:
it may be problematic to overwrite an existent local identifier

it may be better to raise an error if the local identifier (if any) differs from the remote one

I'd suggest to extend the method `getOrSetUUID` in server.go file before initializing the remote storage with a parameter `createIfNotExists`, and set its value in the call with the value of  `!s.Options.RemoteStorageOptions.S3ExternalIdentifier` ...